src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeCustomAttributeData.cs

```diff
@@ -1794,8 +1830,26 @@ internal static AttributeUsageAttribute GetAttributeUsage(RuntimeType decoratedA
                 if (attributeUsageAttribute is not null)
                     throw new FormatException(SR.Format(SR.Format_AttributeUsage, attributeType));
 
-                ParseAttributeUsageAttribute(caRecord.blob, out AttributeTargets targets, out bool inherited, out bool allowMultiple);
-                attributeUsageAttribute = new AttributeUsageAttribute(targets, allowMultiple, inherited);
+                CustomAttributeEncodedArgument.ParseAttributeArguments(
```

Comment:
How does replacing the specialized Parse with the general purpose Parse affect perf?

Reply:
There is some regression here. Opinions on how much latitude?

Reply:
Custom attribute parsing is something that multiple people tried to optimize. For example: https://github.com/dotnet/runtime/issues/44713 and linked issues

Reply:
Creating anything bespoke was going to be annoying. Converted the function to a QCall and added `SuppressGCTransition`. Got back all the time in a simple benchmark.